openquake/hazardlib/gsim/arroyo_2010.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2014-2021 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:'ArroyoEtAl2010SInter'
+"""
+import numpy as np
+from scipy.constants import g
+from scipy.special import exp1
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, SA
+
+def _compute_mean(C, g, ctx):
+    """
+    Compute mean according to equation 8a, page 773.
+    """
+    mag = ctx.mag
+    dis = ctx.rrup
+
+    # computing r02 parameter and the average distance to the fault surface
+    ro2 = 1.4447e-5 * np.exp(2.3026 * mag)
+    avg = np.sqrt(dis ** 2 + ro2)
+
+    # computing fourth term of Eq. 8a, page 773.
+    trm4 = (exp1(C['c4'] * dis) - exp1(C['c4'] * avg)) / ro2
+
+    # computing the mean
+    mean = C['c1'] + C['c2'] * mag + C['c3'] * np.log(trm4)
+        
+    # convert from cm/s**2 to 'g'
+    mean = np.log(np.exp(mean) * 1e-2 / g)
+    return mean
+
+def _get_stddevs(C):
+    """
+    Return standard deviations as defined in table 2, page 776.
+    """
+    stds = np.array([C['s_t'], C['s_e'], C['s_r']])
+    return stds
+
+
+class ArroyoEtAl2010SInter(GMPE):
+    """
+    Implements GMPE developed by Arroyo et al. (2010) for Mexican 
+    subduction interface events and published as:
+
+    Arroyo D., García D., Ordaz M., Mora M. A., and Singh S. K. (2010) 
+    "Strong ground-motion relations for Mexican interplate earhquakes", 
+    J. Seismol., 14:769-785.
+
+    The original formulation predict peak ground acceleration (PGA), in
+    cm/s**2, and 5% damped pseudo-acceleration response spectra (PSA) in 
+    cm/s**2 for the geometric average of the maximum component of the two 
+    horizontal component of ground motion.
+
+    The GMPE predicted values for Mexican interplate events at rock sites
+    (NEHRP B site condition) in the forearc region.
```

Comment:
Perhaps you could add a short note to explain how you created the verification tables.

Reply:
Thank you for your comment. I added a short note as suggested in the code: arroyo_2010_test.py. Please, let me know if it is correct.